src/types.h

```diff
@@ -459,15 +459,14 @@ class SizedType {
   }
   bool IsMapIterableTy() const
   {
-    return !IsMultiOutputMapTy();
+    return !IsMultiKeyMapTy() && (IsCastableMapTy() || !NeedsPercpuMap());
```

Comment:
Honestly (and maybe because it's Monday) but my brain hurts trying to read this.

How about something like this, which is more verbose but requires less thinking:
```
if (IsMultiKeyMapTy()) {
  return false;
}
if (NeedsPercpuMap() && !IsCastableMapTy()) {
  return false;
}
return true;
```

Reply:
Done

Reply:
uh oh - did I give bad logical advice here ?